crates/sparrow-main/src/materialize.rs

```diff
@@ -0,0 +1,192 @@
+        )
+        .await
+        .change_context(Error::Compilation)?;
+
+        let diagnostics = compile_result.fenl_diagnostics.unwrap_or_default();
+        #[allow(clippy::print_stdout)]
+        let plan = if let Some(plan) = compile_result.plan {
+            println!("{diagnostics}");
+            plan
+        } else {
+            error_stack::bail!(Error::InvalidQuery(diagnostics));
+        };
+
+        let s3_helper = S3Helper::new().await;
+
+        // Note: it might be cleaner to create a separate entry point for materialize, but for now it's ok.
+        let result_stream = sparrow_runtime::execute::execute(
+            ExecuteRequest {
+                plan: Some(plan),
+                tables,
+                destination: Some(output_to),
+                limits: None,
+                compute_snapshot_config: None,
+                changed_since: None,
+                final_result_time: None,
+            },
+            s3_helper,
+            self.flight_record_path,
+            FlightRecordHeader::default(),
+        )
+        .await
+        .change_context(Error::Execution)?;
+
+        // For the CLI, we use stdout to print information.
+        #[allow(clippy::print_stdout)]
+        {
+            // Output file vec if the destination is an object store.
+            let mut output_files: Vec<_> = Vec::new();
+
+            let _: Vec<_> = result_stream
+                .inspect_ok(|next| {
+                    let rows_produced = next
+                        .progress
+                        .as_ref()
+                        .expect("progress")
+                        .produced_output_rows;
+                    let destination = next.destination.as_ref().expect("destination");
+                    match &destination.destination {
+                        Some(destination::Destination::ObjectStore(o)) => {
+                            match &o.output_paths {
+                                Some(paths) => {
+                                    paths
+                                        .paths
+                                        .clone()
+                                        .into_iter()
+                                        .for_each(|p| output_files.push(p));
+                                }
+                                _ => (),
+                            };
+                            // TODO: The output paths are empty because we currently don't
+                            // report that a file is being written to until we've closed it.
+                            // Since in a streaming world, we don't close it, we'll indefinitely write
+                            // to a single file.
+                            // We need to:
+                            // 1. Write to multiple output files
+                            // 2. Add to output_files, then log when a new file is produced
+                            println!("{} rows produced so far", rows_produced);
```

Comment:
We log once what file we're writing to the in the beginning, and since this is mainly intended for use in short demos I'm okay with not adding this feature in this PR.